### ISSUE TITLE
Resolved promise resets selection

### ISSUE THREAD

**Issue by AW-UC** [miguelcobain/ember-cli-selectize#93]
I am using `ember-cli-selectize 0.3.8` in `ember-cli 1.13.1` like this:
```
{{ember-selectize
  content=model2.items
  optionValuePath='content.id'
  optionLabelPath='content.name'
  selection=item
  value=item.id}}
```
`model2.items` is a `DS.PromiseArray`. `item` is set via template's controller to `model.item`. What I get is this:

1. Once rendered, you can spot for a short moment, that `item` is pre-selected
2. `model2.items` resolves
3. the selection is reset (cleared)

I workaround this by adding the following to the template's controller:
```
updateSelection: function () {
  this.set('item', this.get('model.item'));
}.observes('model2.items'),
```
Is this the intended behaviour? Any recommendations for improving my code?

**Comment by miguelcobain** [comment]
Why are you using both `selection` and `value` bound to the same `item` property?

**Comment by AW-UC** [comment]
I just thought that is the way to use it: 

* `selection` to define the model
* `value` to define the model's value attribute

I guess `value` is redundant here, then? I removed it. Issue is still there, though.

**Comment by miguelcobain** [comment]
Ah, right. When `model2.items` resolves, can you tell that it contains `item` in it?
Because selectize only allows to be selected an item which *is contained* in `content`.

Also, how does this go if you use:
```hbs
{{ember-selectize
  content=model2.items.content
  optionValuePath='content.id'
  optionLabelPath='content.name'
  selection=item}}
```

Also, any reason for not using `selection=model.item` directly?

**Comment by AW-UC** [comment]
Yes, `model2.items` contains `item`. I am not using `model.item` directly, because that throws:
```
Cannot delegate set('id', (null)) to the 'content' property of object proxy <DS.PromiseObject:ember1042>: its 'content' is undefined.
```
According to http://stackoverflow.com/a/12502466 one is supposed to prevent this, by declaring the property before binding it. That is why I declare:
```
item: Ember.computed.reads('model.item'),
```

**Comment by adjohnson916** [comment]
+1 I'm having a similar issue of the selection being cleared when `content` array refreshes.

**Comment by tschoartschi** [comment]
I'm on version ```0.5.2``` and I'm experiencing a similar issue. Does some one knows how to solve this problem?

**Comment by vasilakisfil** [comment]
I have the same issue as well.. Does anyone know a quick fix on that ?
